File: spring-boot/oidc-client-login-stateless-session-cache/src/main/java/no/acntech/sandbox/controller/ViewController.java
package no.acntech.sandbox.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.servlet.ModelAndView;

@Controller
public class ViewController {

    @GetMapping(path = "/")
    public ModelAndView indexPage() {
        return new ModelAndView("index");
    }

    @GetMapping(path = "/about")
    public ModelAndView aboutPage() {
        return new ModelAndView("about");
    }
}

File: spring-boot-kotlin/grpc/client/src/main/kotlin/no/acntech/sandbox/controller/ViewController.kt
package no.acntech.sandbox.controller

import no.acntech.sandbox.model.Greeting
import no.acntech.sandbox.model.Hello
import no.acntech.sandbox.service.GreetingService
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.servlet.ModelAndView

@Controller
class ViewController(val greetingService: GreetingService) {

    @GetMapping(path = ["/"])
    fun getHomePage(): ModelAndView {
        val mav = ModelAndView("home")
        mav.addObject("hello", Hello())
        return mav
    }

    @PostMapping(path = ["/"])
    fun postHomePage(@ModelAttribute hello: Hello): ModelAndView {
        val greeting: Greeting = greetingService.getGreeting(hello)
        val mav = ModelAndView("home")
        mav.addObject("hello", Hello())
        mav.addObject("greeting", greeting)
        return mav
    }

    @GetMapping(path = ["/about"])
    fun getAboutPage(): ModelAndView {
        return ModelAndView("about")
    }
}